datafusion/physical-expr-common/src/aggregate/mod.rs

```diff
@@ -333,11 +351,24 @@ impl AggregateExpr for AggregateFunctionExpr {
     }
 
     fn create_groups_accumulator(&self) -> Result<Box<dyn GroupsAccumulator>> {
-        self.fun.create_groups_accumulator()
+        let args = AccumulatorArgs {
+            data_type: &self.data_type,
+            schema: &self.schema,
+            ignore_nulls: self.ignore_nulls,
+            sort_exprs: &self.sort_exprs,
+            is_distinct: self.is_distinct,
+            input_type: &self.input_type,
+            args_num: self.args.len(),
+            name: &self.name,
+        };
+        self.fun.create_groups_accumulator(args)
     }
 
     fn order_bys(&self) -> Option<&[PhysicalSortExpr]> {
-        (!self.ordering_req.is_empty()).then_some(&self.ordering_req)
+        if self.fun.has_ordering_requirements() && !self.ordering_req.is_empty() {
```

Comment:
Even if `has_ordering_requirements` is not introduced in this PR. I think, with the `order_sensitivity` API. This API is redundant.  I think, it is better to remove this API (in this Pr or in subsequent ones).

Reply:
I agree. We can returns `ordering_req` if sensitivity is not `AggregateOrderSensitivity::Insensitive`

Reply:
I agree. We can returns None for `AggregateOrderSensitivity::Insensitive`, `ordering_req` if others

Reply:
I agree that, for most of the aggregate functions, `AggregateOrderSensitivity::Insensitive` is the correct behavior. However, I think the safest default choice is `AggregateOrderSensitivity::HardRequirement`. Also, as long as there is no requirement, default is not important. Hence, I think we can be strict in this choice.